### PR TITLE
Find products by partial name added

### PR DESCRIPTION
**Pull request by GeorgeT88**

### CODE REVIEW

File: src/main/java/com/boot/product/controller/ProductController.java [hunk 1]
```diff
@@ -84,6 +84,16 @@ public ResponseEntity<ProductDTO> findProductBySlug(@Size(min = 3, max = 30, mes
         return new ResponseEntity<>(product, HttpStatus.OK);
     }
 
+    @GetMapping("/partialName/{partialName}")
+    @ResponseBody
+    public ResponseEntity<List<String>> findProductsByPartialName(@Size(min = 3, max = 30, message = "Product Name size has to be between 3 and 30 characters!") @PathVariable("partialName") String partialName)
```
Comment (florinv23) [inline src/main/java/com/boot/product/controller/ProductController.java]: I wouldn't see the input string as a product name so maybe the error message might not be accurate

File: src/main/java/com/boot/product/controller/ProductController.java [hunk 2]
```diff
@@ -84,6 +84,16 @@ public ResponseEntity<ProductDTO> findProductBySlug(@Size(min = 3, max = 30, mes
         return new ResponseEntity<>(product, HttpStatus.OK);
     }
 
+    @GetMapping("/partialName/{partialName}")
```
Comment (florinv23) [inline src/main/java/com/boot/product/controller/ProductController.java]: I wouldn't put the search string as a pathVariable but as a query param.
I see that the emag call looks like:
/search-suggester?source_id=7&query=masina&shop_id=1

File: src/main/java/com/boot/product/repository/ProductRepository.java [hunk 3]
```diff
@@ -21,6 +21,8 @@ public interface ProductRepository extends JpaRepository<Product, Long> {
 
     Product findBySlug(String slug);
 
+    List<Product> findByNameContainingIgnoreCase(String name);
```
Comment (florinv23) [inline src/main/java/com/boot/product/repository/ProductRepository.java]: this query method returns all product data. you should create one that retrieves only the product name

File: src/main/java/com/boot/product/service/ProductService.java [hunk 4]
```diff
@@ -118,6 +118,14 @@ public ProductDTO getProductBySlug(String slug, Boolean includeInactive) {
         return productEntityToDto(product);
     }
 
+    public List<String> getProductsByPartialName(String name) {
+
+        log.info("getProductsByPartialText - process started");
+
+        return productRepository.findByNameContainingIgnoreCase(name).stream().map(Product::getName)
+                .collect(Collectors.toList()).stream().limit(8).collect(Collectors.toList());
```
Comment (florinv23) [inline src/main/java/com/boot/product/service/ProductService.java]: this is where you limit the number of products. so basically you retrieve all matching products from db and limit the returned ones here

File: src/main/java/com/boot/product/controller/ProductController.java (hunk 1 above)
Comment (GeorgeT88) [reply src/main/java/com/boot/product/controller/ProductController.java]: well, its a partial product name..
Any suggestions? I couldn't find a better one :)

File: src/main/java/com/boot/product/service/ProductService.java (hunk 4 above)
Comment (GeorgeT88) [reply src/main/java/com/boot/product/service/ProductService.java]: yea, should i change something here?

File: src/main/java/com/boot/product/repository/ProductRepository.java (hunk 3 above)
Comment (GeorgeT88) [reply src/main/java/com/boot/product/repository/ProductRepository.java]: not shure how to do this, cand you please link some reference for it?

File: src/main/java/com/boot/product/service/ProductService.java (hunk 4 above)
Comment (florinv23) [reply src/main/java/com/boot/product/service/ProductService.java]: yes. you could pass a pageable object like I've used on the get products endpoint but a pageable that only the **size** is taken from the request parameters. **page** should be always 0, **sort** criteria will always be name and the **direction** ascending.

File: src/main/java/com/boot/product/repository/ProductRepository.java (hunk 3 above)
Comment (florinv23) [reply src/main/java/com/boot/product/repository/ProductRepository.java]: you can use the query annotation
@Query("select name from Product where name like '%:names%'")

File: src/main/java/com/boot/product/controller/ProductController.java (hunk 1 above)
Comment (florinv23) [reply src/main/java/com/boot/product/controller/ProductController.java]: Query string? or Partial product name?

**Pull request merged**